kahuna/public/js/edits/image-editor.js

```diff
@@ -73,6 +75,18 @@ imageEditor.controller('ImageEditorCtrl',
         ctrl.error = true;
     }
 
+    mediaApi.canDelete(ctrl.image).then(deletable => {
+        ctrl.canBeDeleted = deletable;
+    });
+
+
+    ctrl.onDeleteSuccess = () => {
+        $state.go('upload', {}, {reload: true});
```

Comment:
Suggested emitting a notification, similar to image-updated, which the image upload service could listen to to update its state (remove deleted images).